src/libraries/System.Threading.RateLimiting/src/System/Threading/RateLimiting/ChainedRateLimiter.cs

```diff
@@ -0,0 +1,191 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace System.Threading.RateLimiting
+{
+    /// <summary>
+    /// Acquires leases from rate limiters in the order given. If a lease fails to be acquired (throwing or IsAcquired == false)
+    /// then the already acquired leases are disposed in reverse order and the failing lease is returned or the exception is thrown to user code.
+    /// </summary>
+    internal sealed partial class ChainedRateLimiter : RateLimiter
+    {
+        private readonly RateLimiter[] _limiters;
+        private bool _disposed;
+
+        public ChainedRateLimiter(RateLimiter[] limiters)
+        {
+            _limiters = (RateLimiter[])limiters.Clone();
+        }
+
+        public override RateLimiterStatistics? GetStatistics()
+        {
+            ThrowIfDisposed();
+            long lowestAvailablePermits = long.MaxValue;
+            long currentQueuedCount = 0;
+            long totalFailedLeases = 0;
+            long innerMostSuccessfulLeases = 0;
+            foreach (RateLimiter limiter in _limiters)
+            {
+                if (limiter.GetStatistics() is { } statistics)
+                {
+                    if (statistics.CurrentAvailablePermits < lowestAvailablePermits)
+                    {
+                        lowestAvailablePermits = statistics.CurrentAvailablePermits;
+                    }
+                    currentQueuedCount += statistics.CurrentQueuedCount;
+                    totalFailedLeases += statistics.TotalFailedLeases;
+                    innerMostSuccessfulLeases = statistics.TotalSuccessfulLeases;
+                }
+            }
+            return new RateLimiterStatistics()
+            {
+                CurrentAvailablePermits = lowestAvailablePermits,
+                CurrentQueuedCount = currentQueuedCount,
+                TotalFailedLeases = totalFailedLeases,
+                TotalSuccessfulLeases = innerMostSuccessfulLeases,
+            };
+        }
+
+        public override TimeSpan? IdleDuration
+        {
+            get
+            {
+                ThrowIfDisposed();
+                TimeSpan? lowestIdleDuration = null;
+                foreach (RateLimiter limiter in _limiters)
+                {
+                    if (limiter.IdleDuration is { } idleDuration)
+                    {
+                        if (lowestIdleDuration is null || idleDuration < lowestIdleDuration)
+                        {
+                            lowestIdleDuration = idleDuration;
+                        }
+                    }
+                }
+                return lowestIdleDuration;
```

Comment:
throughout the code, please fix the spacing, e.g.
```suggestion
                ThrowIfDisposed();

                TimeSpan? lowestIdleDuration = null;
                foreach (RateLimiter limiter in _limiters)
                {
                    if (limiter.IdleDuration is { } idleDuration)
                    {
                        if (lowestIdleDuration is null || idleDuration < lowestIdleDuration)
                        {
                            lowestIdleDuration = idleDuration;
                        }
                    }
                }

                return lowestIdleDuration;
```

Reply:
Spacing reworked in all modified files